ethereum/contracts/ETHApp.sol

```diff
@@ -80,17 +74,11 @@ contract ETHApp {
         emit Unlocked(_sender, _recipient, _amount);
     }
 
-    // SCALE-encode payload
-    function encodePayload(OutboundPayload memory payload)
+    function encodeCall(address _sender, bytes32 _recipient, uint256 _amount)
         private
         pure
         returns (bytes memory)
     {
-        return
-            abi.encodePacked(
-                payload.sender,
-                payload.recipient,
-                payload.amount.toBytes32LE()
-            );
+        return abi.encodePacked(CALL_INDEX, _sender, _recipient, _amount.toBytes32LE());
```

Comment:
do you have a guess for how much this adds to the cost of the transaction in USD?

Reply:
You mean the new SCALE-encoding vs older ABI-encoding?

Well I ran a benchmark and the SCALE-encoding in this PR adds 6.64% to the gas used (76240 vs 81307) by the `lock()` transaction. At current gas prices (average confirmation times) that adds an extra $1.77 to the cost of the transaction, coming up to a total of $26.64879 according to https://ethgasstation.info/calculatorTxV.php.

Reply:
Though gas prices are unusually high today, roughly 50% more than the average over the past 4 months.

Reply:
you mean 6% just for adding the  CALL_INDEX parameter? or 6% in total compared to abi encoding?

Reply:
the latter, 6% in total compared to abi encoding.

Reply:
I've further optimized the SCALE-encoding of `uint256`, so its now about 21.5% faster than the previous algorithm. So total  overhead should now only be about 5% instead of 6%.